src/librustc_middle/ty/query/values.rs

```diff
@@ -21,9 +19,15 @@ impl<'tcx> Value<'tcx> for &'_ TyS<'_> {
     }
 }
 
-impl<'tcx> Value<'tcx> for ty::SymbolName {
-    fn from_cycle_error(_: TyCtxt<'tcx>) -> Self {
-        ty::SymbolName { name: Symbol::intern("<error>") }
+impl<'tcx> Value<'tcx> for ty::SymbolName<'_> {
+    fn from_cycle_error(tcx: TyCtxt<'tcx>) -> Self {
+        // SAFETY: This is never called when `Self` is not `SymbolName<'tcx>`.
+        // FIXME: Represent the above fact in the trait system somehow.
```

Comment:
~~Why can't you write `ty::SymbolName<'tcx>`? That's why `Value` has a `'tcx` parameter.~~

Oh this is a specialization issue. IMO it should be noted as such rather then left as a vague `FIXME` (cc @matthewjasper).

Reply:
What exactly would you like me to write in the comment?

Reply:
I hope @matthewjasper can come up with a better comment. Otherwise it's probably fine as-is.
Although I kind of doubt this `from_cycle_error` implementation is needed at all.